lib/miq_automation_engine/models/miq_ae_git.rb

```diff
@@ -0,0 +1,328 @@
+require 'rugged'
+
+class MiqAeGit
+  attr_accessor :name, :email, :base_name
+  ENTRY_KEYS = [:path, :dev, :ino, :mode, :gid, :uid, :ctime, :mtime]
+  DEFAULT_FILE_MODE = 0100644
+  LOCK_REFERENCE = 'refs/locks'
+
+  def initialize(options = {})
+    raise ArgumentError, "Must specify path" unless options.key?(:path)
+    @path          = options[:path]
+    @email         = options[:email]
+    @name          = options[:name]
+    @bare          = options[:bare]
+    @commit_sha    = options[:commit_sha]
+    @password      = options[:password]
+    @ssl_no_verify = options[:ssl_no_verify] || false
+    @remote_name   = 'origin'
+    @cred          = Rugged::Credentials::UserPassword.new(:username => @name,
+                                                           :password => @password)
+    @base_name     = File.basename(@path)
+    return clone(options[:url]) if options[:url]
```

Comment:
This return doesn't do anything...so why does it there?  Or are you trying to actually do:

```ruby
if options[:url]
  clone(options[:url])
elsif options[:new]
  create_repo
else
  open_repo
end
```